packages/block-library/src/social-link/index.php

```diff
@@ -32,9 +32,9 @@ function render_block_core_social_link( $attributes, $content, $block ) {
 		return '';
 	}
 
-	$attribute = '';
+	$rel_target_attributes = '';
 	if ( $open_in_new_tab ) {
-		$attribute = 'rel="noopener nofollow" target="_blank"';
+		$rel_target_attributes = 'rel="' . esc_attr( 'noopener nofollow' ) . '" target="' . esc_attr( '_blank' ) . '"';
```

Comment:
Since these are hardcoded and there's no user-input involved, there's no reason to escape them.

Reply:
Yeh I take your point.

To give you an idea of my thinking, I'm trying to avoid situations where we have hard coded variables that are not escaped at point of output.

Whilst these variables might contain hardcoded content _now_ there's nothing to say they will stay that way. What if developer A comes along and makes the `rel` attribute configurable from the block. What if they simply put the attribute provided by the user into the variable above and forget to escape? What happens then?

By being a little more verbose with escaping now, we avoid potential future XSS issues being introduced. For example if Developer A were to come along to the code I've suggested here there's no danger they will forget to escape the `rel` attribute. Personally I think the trade off is worth it.

This is a unusual use case because we aren't actually escaping as close to the building of the HTML as I"d like due to the conditional rendering of the x2 attributes. Ideally we'd simply escape the output in the HTML template rather than do it when we're building the variable.

Don't get me wrong, I understand this is pretty verbose and might seem like over doing things. However in my experience it's better to be safe than sorry.

Reply:
If you're not happy with this we can just close this PR out.

Reply:
I'm with Ari on this one. There's no need to escape hardcoded strings.

Reply:
Actually reflecting on this again, I've now _removed_ the variable and so these strings are inline. Therefore I agree why are we escaping? 

My main concern is variables not being escaped. I'll make the change.

Reply:
Thanks, Dave.

Reply:
When I started making the change I noticed that the `rel` attribute is in fact _not_ inlined. I still have reservations but they are relatively minor so I've removed the escaping as per the majority opinion.